app/uk/gov/hmrc/cgtpropertydisposalsfrontend/services/UKAddressLookupService.scala

```diff
@@ -87,7 +86,22 @@ class UKAddressLookupServiceImpl @Inject() (
     postcode: Postcode,
     filter: Option[String]
   ): Either[String, AddressLookupResult] = {
-    def toAddress(a: RawAddress): Either[String, Address] = {
+
+    def validateAddressFields(address: UkAddress): Boolean = {
+      val addressLineRegex = "^[A-Za-z0-9 \\-,.&'/]{0,35}$".r.pattern.asPredicate()
```

Comment:
you could put these regex outside of the function to avoid creating a load a temporary vlaues which are the same thing everything this function is being called. Since it's the same regex that is used in `Address` and `Postcode`, you could possible make those available in those objects to reuse here